packages/dataviews/src/components/dataviews/index.tsx

```diff
@@ -177,7 +191,14 @@ function DataViews< Item >( {
 			} }
 		>
 			<div className="dataviews-wrapper" ref={ containerRef }>
-				{ children || defaultUI }
+				{ children ?? (
+					<DefaultUI
+						header={ header }
+						search={ search }
+						searchLabel={ searchLabel }
+						isShowingFilter={ isShowingFilter }
```

Comment:
Outside "header" which is specific to the "default ui" rendering, the other props could be something that get moved into context (and not rely on prop drilling).

Reply:
same PR?

Reply:
I'm not sure actually whether it's a good thing or not, It's up for debate :P

Reply:
`isShowingFilter` is already available in the context, which makes sense since it enabled the decoupling of the filter components.

The other props seem tightly coupled to the default UI, and I’m not sure it makes sense to move them into the context.

That said, let’s go ahead and pick `isShowingFilter` from the context.

Reply:
> I wonder if we should remove or deprecate the header prop, now that free composition is supported.

I'd be in favor of that, but since it's used in Gutenberg and would require a migration there later, I think let's just keep it for now, it's not that impactful.

Reply:
`header` is also used in other products like Jetpack, so definitely `deprecate` first, if we are to remove it eventually.